server/common/utils/index.js

```diff
@@ -53,29 +54,83 @@ const responseWithItem = (item, userId) => {
 const responseWithCohorts = (cohorts, userId) =>
   _map(cohorts, ({ _doc }) => {
     const { favIds, ...rest } = _doc;
+
     return {
       ...rest,
       isFavourite: isUserFavourite(favIds, userId)
     };
   });
 
+const checkIfOwner = (ownerIds, userId) =>
+  ownerIds.indexOf(userId.toString()) > -1;
+
 const responseWithMembers = (users, ownerIds) =>
   users.map(user => ({
     ...user._doc,
-    isOwner: ownerIds.indexOf(user._doc._id.toString()) > -1
+    isOwner: checkIfOwner(ownerIds, user._doc._id)
   }));
 
+const checkIfCohortMember = (cohort, userId) => {
+  if (cohort) {
+    const { memberIds, ownerIds } = cohort;
+    const convertedUserId = ObjectId(userId);
+    return [...memberIds, ...ownerIds].some(id => id.equals(convertedUserId));
+  }
+
+  return false;
+};
+
+const checkIfGuest = (data, userId) => {
+  if (data) {
+    return data.constructor === Array
+      ? !data.some(member => member._id.equals(userId))
+      : ![...data.memberIds, ...data.ownerIds].some(id => id.equals(userId));
+  }
+
+  return true;
+};
+
 const responseWithMember = (data, ownerIds) => {
   const { avatarUrl, displayName, newMemberId } = data;
+
   return {
     _id: newMemberId,
     avatarUrl,
     displayName,
-    isOwner: ownerIds.indexOf(newMemberId.toString()) > -1
+    isOwner: checkIfOwner(ownerIds, newMemberId)
   };
 };
 
+const responseWithListMember = (data, ownerIds, cohort) => {
+  const { avatarUrl, displayName, newMemberId } = data;
+
+  return {
+    _id: newMemberId,
+    avatarUrl,
+    displayName,
+    isGuest: checkIfGuest(cohort, newMemberId),
+    isOwner: checkIfOwner(ownerIds, newMemberId)
+  };
+};
+
+const responseWithListMembers = (users, ownerIds, cohortMembers) =>
+  users.map(user => ({
+    ...user._doc,
+    isOwner: checkIfOwner(ownerIds, user._doc._id),
+    isGuest: checkIfGuest(cohortMembers, user._doc._id)
+  }));
+
+const uniqueMembers = (cohortMembers, listMembers) =>
```

Comment:
what this function does?
Maybe add some small comment above this, cause it's not very clear

Reply:
All cohort's members are members of public list. public list may contain some of them in it's ownerIds but additionally it can contain guestIds which are not cohort's. So we have to merge those arrays (to have all public list members) and remove duplicated user's ids.

Reply:
add some describing comment above this function